iceoryx_utils/test/moduletests/test_atomic_relocatable_ptr.cpp

```diff
@@ -156,19 +156,20 @@ TEST_F(AtomicRelocatablePointer_test, dereferenceOperator)
 // copied location
 TEST_F(AtomicRelocatablePointer_test, memoryRelocation)
 {
-    Memory<1024> memory;
+    constexpr uint64_t ALIGNMENT_OF_PTR {alignof(Ptr<byte_t>)};
+    Memory<1024, ALIGNMENT_OF_PTR> memory;
     memory[1000] = 37;
     // EXPECT_EQ(memory[1000], 37);
 
-    Ptr<byte_t>* rp = new (memory.ptr(10)) Ptr<byte_t>(memory.ptr(1000));
+    Ptr<byte_t>* rp = new (memory.ptr(ALIGNMENT_OF_PTR)) Ptr<byte_t>(memory.ptr(1000));
 
     // we have constructed a relocatable pointer at adress 10 in memory which points
     // to address at memory location 1000 which holds value 37
     EXPECT_EQ(*rp, memory.ptr(1000));
     EXPECT_EQ(**rp, 37);
 
     // copy this memory to a new destination which we then set to 0
-    Memory<1024> dest;
+    Memory<1024, ALIGNMENT_OF_PTR> dest;
```

Comment:
Indeed, when both memory locations (memory and dest) are aligned as the rp which is the first actual data structure inside it the test should be robust against failures due to misaligned data (either due to emplacement new or unaligned memcpy).